src/gt4py/next/ffront/foast_passes/type_deduction.py

```diff
@@ -304,9 +304,25 @@ def visit_ScanOperator(self, node: foast.ScanOperator, **kwargs) -> foast.ScanOp
                 f"be an arithmetic type or a logical type or a composite of arithmetic and logical types.",
             )
         new_definition = self.visit(node.definition, **kwargs)
+        new_def_type = new_definition.type
+        state_type = list(new_def_type.pos_or_kw_args.values())[0]
```

Comment:
```suggestion
        carry_type = list(new_def_type.pos_or_kw_args.values())[0]
```
We commonly use `carry` as a phrase for this argument (https://github.com/GridTools/gt4py/blob/444d660293db645f0a7d5032398ac64946dd6c04/src/gt4py/next/ffront/decorator.py#L656). The name is borrowed from Jax's scan pass.